[2025-06-11T17:45:42.412+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T15:45:38.079269+00:00 [queued]>
[2025-06-11T17:45:42.434+0200] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T15:45:38.079269+00:00 [queued]>
[2025-06-11T17:45:42.434+0200] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-06-11T17:45:42.435+0200] {taskinstance.py:1280} INFO - Starting attempt 1 of 4
[2025-06-11T17:45:42.435+0200] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-06-11T17:45:42.462+0200] {taskinstance.py:1300} INFO - Executing <Task(BashOperator): pyspark_train_classifier_model> on 2025-06-11 15:45:38.079269+00:00
[2025-06-11T17:45:42.466+0200] {standard_task_runner.py:55} INFO - Started process 176591 to run task
[2025-06-11T17:45:42.469+0200] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'agile_data_science_batch_prediction_model_training', 'pyspark_train_classifier_model', 'manual__2025-06-11T15:45:38.079269+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/setup.py', '--cfg-path', '/tmp/tmpbvdqp1st']
[2025-06-11T17:45:42.470+0200] {standard_task_runner.py:83} INFO - Job 15: Subtask pyspark_train_classifier_model
[2025-06-11T17:45:42.602+0200] {task_command.py:388} INFO - Running <TaskInstance: agile_data_science_batch_prediction_model_training.pyspark_train_classifier_model manual__2025-06-11T15:45:38.079269+00:00 [running]> on host l090.lab.dit.upm.es
[2025-06-11T17:45:42.735+0200] {taskinstance.py:1507} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=agile_data_science_batch_prediction_model_training
AIRFLOW_CTX_TASK_ID=pyspark_train_classifier_model
AIRFLOW_CTX_EXECUTION_DATE=2025-06-11T15:45:38.079269+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-06-11T15:45:38.079269+00:00
[2025-06-11T17:45:42.737+0200] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2025-06-11T17:45:42.738+0200] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\nspark-submit --master local[4]   /home/monica.fernandez/practica_creativa//resources/train_spark_mllib_model.py   /home/monica.fernandez/practica_creativa/']
[2025-06-11T17:45:42.753+0200] {subprocess.py:86} INFO - Output:
[2025-06-11T17:45:45.026+0200] {subprocess.py:93} INFO - 25/06/11 17:45:45 WARN Utils: Your hostname, l090 resolves to a loopback address: 127.0.1.1; using 138.4.31.90 instead (on interface enp3s0)
[2025-06-11T17:45:45.030+0200] {subprocess.py:93} INFO - 25/06/11 17:45:45 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/resources/train_spark_mllib_model.py", line 6, in <module>
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -     import mlflow
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/__init__.py", line 44, in <module>
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -     from mlflow import (
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/artifacts/__init__.py", line 11, in <module>
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -     from mlflow.entities.file_info import FileInfo
[2025-06-11T17:45:45.969+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/entities/__init__.py", line 6, in <module>
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -     from mlflow.entities.assessment import (
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/entities/assessment.py", line 12, in <module>
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -     from mlflow.entities.assessment_error import AssessmentError
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/entities/assessment_error.py", line 6, in <module>
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -     from mlflow.utils.annotations import experimental
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/utils/__init__.py", line 10, in <module>
[2025-06-11T17:45:45.970+0200] {subprocess.py:93} INFO -     from mlflow.utils.pydantic_utils import IS_PYDANTIC_V2_OR_NEWER  # noqa: F401
[2025-06-11T17:45:45.971+0200] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-06-11T17:45:45.971+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/mlflow/utils/pydantic_utils.py", line 3, in <module>
[2025-06-11T17:45:45.971+0200] {subprocess.py:93} INFO -     import pydantic
[2025-06-11T17:45:45.971+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/pydantic/__init__.py", line 413, in <module>
[2025-06-11T17:45:45.972+0200] {subprocess.py:93} INFO -     _getattr_migration = getattr_migration(__name__)
[2025-06-11T17:45:45.974+0200] {subprocess.py:93} INFO -                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-06-11T17:45:45.974+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/pydantic/_migration.py", line 260, in getattr_migration
[2025-06-11T17:45:45.974+0200] {subprocess.py:93} INFO -     from .errors import PydanticImportError
[2025-06-11T17:45:45.974+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/pydantic/errors.py", line 9, in <module>
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO -     from typing_inspection.introspection import Qualifier
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/typing_inspection/introspection.py", line 14, in <module>
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO -     from . import typing_objects
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO -   File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/typing_inspection/typing_objects.py", line 19, in <module>
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO -     from typing_extensions import LiteralString, TypeAliasType, TypeIs, deprecated
[2025-06-11T17:45:45.975+0200] {subprocess.py:93} INFO - ImportError: cannot import name 'TypeIs' from 'typing_extensions' (/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/typing_extensions.py)
[2025-06-11T17:45:46.021+0200] {subprocess.py:93} INFO - 25/06/11 17:45:46 INFO ShutdownHookManager: Shutdown hook called
[2025-06-11T17:45:46.022+0200] {subprocess.py:93} INFO - 25/06/11 17:45:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-5a915238-9d24-4d82-9f43-88fc4991d73e
[2025-06-11T17:45:46.065+0200] {subprocess.py:97} INFO - Command exited with return code 1
[2025-06-11T17:45:46.120+0200] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/monica.fernandez/practica_creativa/venv-airflow/lib/python3.11/site-packages/airflow/operators/bash.py", line 196, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-06-11T17:45:46.124+0200] {taskinstance.py:1318} INFO - Marking task as UP_FOR_RETRY. dag_id=agile_data_science_batch_prediction_model_training, task_id=pyspark_train_classifier_model, execution_date=20250611T154538, start_date=20250611T154542, end_date=20250611T154546
[2025-06-11T17:45:46.143+0200] {standard_task_runner.py:100} ERROR - Failed to execute job 15 for task pyspark_train_classifier_model (Bash command failed. The command returned a non-zero exit code 1.; 176591)
[2025-06-11T17:45:46.177+0200] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-06-11T17:45:46.257+0200] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
